sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -86,6 +86,7 @@ class SparkSession private(
 
   private[sql] def this(sc: SparkContext) {
     this(sc, None, None, new SparkSessionExtensions)
+    SparkSession.applyExtensionsFromConf(sc.getConf, this.extensions)
```

Comment:
Let's add some comments why this is only here in this constructor. It might look weird why this constructor specifically requires to run `applyExtensionsFromConf` alone.